components/ws-daemon/cmd/run.go

```diff
@@ -128,7 +129,21 @@ var runCmd = &cobra.Command{
 			log.WithError(err).Fatal("cannot start daemon")
 		}
 
-		go config.Watch(configFile, dmn.ReloadConfig)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		err = watch.File(ctx, configFile, func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+			defer cancel()
+
+			err := dmn.ReloadConfig(ctx, &cfg.Daemon)
```

Comment:
I don't see where the config is actually re-read from disk. The prior implementation had a `reloadConfig()` call in the `config` package to that end.

Reply:
Fixed